### ISSUE TITLE
Randomized Smoothing Wrapper

### ISSUE THREAD

**Issue by keykholt** [Trusted-AI/adversarial-robustness-toolbox#735]
**Is your feature request related to a problem? Please describe.**
In prior versions of ART, randomized smoothing was implemented as a ClassifierWrapper. In the current version (1.4), it has been re-implemented as an estimator instance. While this new implementation simplifies the process for creating smoothed classifiers, this is only true if the ML framework has a definition. Currently, only smoothed classifiers for pytorch and tensorflow can be created.

**Describe the solution you'd like**
I would like to see the Randomized Smoothing wrapper be returned in order to allow users to wrap a classifier created in any ML framework, even if it doesn't have an implementation in estimators.classifier.

**Describe alternatives you've considered**
One alternative is to define the randomized smoothing estimator class more concretely. It appears the main difference between the pytorch and tensorflow implementation the definition of the loss_gradient. Other than that, predict and fit are the same. If so, why not just push predict/fit into randomizedsmoothing.py? The framework specific implementation can provide a definition for the loss gradient if a user needs that functionality. Otherwise, they can define the RS classifier using the base class in randomizedsmoothing.py. 

Since what is passed to RandomizedSmoothing class is a framework specific model, rather than an ART estimator, there may be an issue figuring out which predict/fit function to call. As such, you could either:
1. Check the type of the model object and infer the framework. After which, you call the correct ART estimator
2. Have the the framework type be a value that is passed to RS so you don't have to infer the framework
3. Require that the model have fit and predict defined. This solution could be useful if the user passes a model in a framework that ART doesn't expect (i.e. a blackbox approach to RS).

**Comment by keykholt** [comment]
Another alternative solution would be to re-introduce the wrapper, but also add the loss gradient to the wrapper definition. Maybe during initialization, you check the type of the model object to infer the framework. If the model belongs to a framework that you have a loss gradient defined for (e.g. pytorch and tensorflow), then you can use the definitions if loss_gradient is ever called.

**Comment by beat-buesser** [comment]
Hi @keykholt 

Thank you very much for your feature request. I think it could be useful to have support for Randomised Smoothing for models of other frameworks than TensorFlow 2.x and PyTorch. We are aiming to abandon and remove the `art.wrappers` module in the near future completely in favor of the new `art.estimators` module for reasons like maintainability, allowing framework-specific implementations for performance reason and future extensibility like full support for expectation over transformation.

If you would like to implement a framework-independent version of Randmosed Smoothing you would have to implement it as a Numpy-based version in `art.estimators.certification.randomized_smoothing.numpy.py` next to the framework-specific implementaitons.
